### PR TITLE
Nick/image cache options

### PR DESCRIPTION
**Pull request by nwadams**
preferentially remove images from cache
increase cache size to 15MB

### CODE REVIEW

File: src/com/android/volley/toolbox/DiskBasedCache.java [hunk 1]
```diff
@@ -280,6 +281,11 @@ private void pruneIfNeeded(int neededSpace) {
             }
         }
 
+        // we weren't able to remove enough images to find space
+        if (mTotalSize + neededSpace >= mMaxCacheSizeInBytes * HYSTERESIS_FACTOR && pruneOnlyImages) {
+            pruneIfNeeded(neededSpace, false);
```
Comment (lxdvs) [inline src/com/android/volley/toolbox/DiskBasedCache.java:285]: recursion!

File: src/com/android/volley/toolbox/DiskBasedCache.java [hunk 2]
```diff
@@ -246,7 +247,7 @@ public File getFileForKey(String key) {
      * Prunes the cache to fit the amount of bytes specified.
      * @param neededSpace The amount of bytes we are trying to fit into the cache.
      */
-    private void pruneIfNeeded(int neededSpace) {
+    private void pruneIfNeeded(int neededSpace, boolean pruneOnlyImages) {
```
Comment (lxdvs) [inline src/com/android/volley/toolbox/DiskBasedCache.java]: s/pruneOnlyImages/preferImagePruning

File: src/com/android/volley/toolbox/DiskBasedCache.java [hunk 3]
```diff
@@ -52,10 +53,10 @@
     private final File mRootDirectory;
 
     /** The maximum size of the cache in bytes. */
-    private final int mMaxCacheSizeInBytes;
+    private final long mMaxCacheSizeInBytes;
 
     /** Default maximum disk usage in bytes. */
-    private static final int DEFAULT_DISK_USAGE_BYTES = 5 * 1024 * 1024;
+    private static final long DEFAULT_DISK_USAGE_BYTES = 15L * 1024 * 1024;
```
Comment (lxdvs) [inline src/com/android/volley/toolbox/DiskBasedCache.java]: compromise on 10?

File: src/com/android/volley/toolbox/DiskBasedCache.java (hunk 2 above)
Comment (nwadams) [inline src/com/android/volley/toolbox/DiskBasedCache.java]: +1

File: src/com/android/volley/toolbox/DiskBasedCache.java (hunk 3 above)
Comment (nwadams) [inline src/com/android/volley/toolbox/DiskBasedCache.java]: We can do 10. I think disk space is cheap though. Let's do 10

**Comment by lxdvs** [comment]
+1

**Pull request merged**